syncer/syncer.go

```diff
@@ -1404,16 +1436,37 @@ func (s *Syncer) syncDML(
 
 // Run starts running for sync, we should guarantee it can rerun when paused.
 func (s *Syncer) Run(ctx context.Context) (err error) {
-	tctx := s.tctx.WithContext(ctx)
+	runCtx, runCancel := context.WithCancel(context.Background())
+	tctx := s.tctx.WithContext(runCtx)
 
 	defer func() {
 		if s.done != nil {
 			close(s.done)
 		}
 	}()
 
+	go func() {
+		<-ctx.Done()
+		tctx.L().Info("received subtask's done")
+		if s.isTransactionEnd.Load() {
+			tctx.L().Info("the last job is transaction end, done directly")
+			runCancel()
+			return
+		}
+		waitTime := 10 * time.Second
+		s.waitXIDJob = waiting
+		tricker := time.NewTicker(waitTime)
+		defer tricker.Stop()
+		select {
+		case <-runCtx.Done():
+			tctx.L().Info("received syncer's done")
+		case <-tricker.C:
```

Comment:
```suggestion
		case <- time.After(10 * time.Second):
```
only need to wait once, so no need to explicitly create a ticker